util/topgen.py

```diff
@@ -480,10 +480,17 @@ def _get_pinmux_params(top: ConfigT) -> ParamsT:
                   pinmux["io_counts"]["dedicated"]["inputs"] +
                   pinmux["io_counts"]["dedicated"]["outputs"])
 
+    tap_strap0_idx = pinmux.get('tap_strap0_idx', 0)
+
     # Generation with zero MIO/DIO pads is currently not supported.
     assert (n_mio_pads > 0)
     assert (n_dio_pads > 0)
 
+    if tap_strap0_idx >= n_mio_pads:
```

Comment:
This is now pulling in an Earlgrey specific thing into the tooling. We should find a way to avoid that and provide the necessary information from the HJSONs and maybe ipgen parameters.

Reply:
Hmm, I think this has the same response as the previous note. I suspect the right thing to do is to set `tap_strap0_idx` to `-1` for Darjeeling (and then this code is completely reasonable, although it would be worth me adding a couple of documentation comments to explain what's going on)

Is that right?

Reply:
Can this check be moved to the parametrized pinmux template?